### PR TITLE
add ESLint support

### PR DESCRIPTION
**Pull request by danbev**
JIRA:
https://issues.jboss.org/browse/FH-3117

### CODE REVIEW

**Comment by danbev** [comment]
@aidenkeating Would you mind taking a look?

File: app.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 var $fh = require('fh-js-sdk');
 
-
+/* eslint-disable */
 console.log('TEST', $fh.sync);
 console.log('window.location.href', window.location.href);
```
Comment (aidenkeating) [inline app.js]: I wonder if these `console.log` statements should be removed?

File: application.js [hunk 2]
```diff
@@ -27,7 +27,8 @@ app.use(mbaasExpress.errorHandler());
 var port = process.env.FH_PORT || process.env.OPENSHIFT_NODEJS_PORT || 8001;
 var host = process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
 var server = app.listen(port, host, function() {
-  console.log("App started at: " + new Date() + " on port: " + port); 
+  // eslint-disable-next-line
+  console.log("App started at: " + new Date() + " on port: " + port);
```
Comment (aidenkeating) [inline application.js]: Same with this, I don't think this is needed.

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "no-spaced-func": "error",
+    "no-trailing-spaces": "error",
+    "semi": ["error", "always"],
+    "space-before-blocks": "error",
+    "space-before-function-paren": ["error", "never"],
+    "keyword-spacing": ["error"],
+    "curly": ["error", "all"]
+  },
+  "globals": {
+    "$fh": true,
```
Comment (aidenkeating) [inline .eslintrc]: This is more of a selfish change as my editor is saying this file isn't valid JSON but should the env values here be double quoted, and this final comma removed

**Review by aidenkeating — APPROVED** [review]
Looks good to me. A few minor comments 👍

File: .eslintrc (hunk 3 above)
Comment (danbev) [reply .eslintrc]: all valid and I've fixed them now. I'll rebase this with upstream/master and make sure the CI tests pass

**Comment by danbev** [comment]
Landed in 14e14b175f5d25b38387cf584caf9b1275d1213f

**Pull request merged**